### PR TITLE
accept more range types for std.path.*trimSeparators()

### PR DESCRIPTION
**Pull request by WalterBright**
Accept input and bidirectional ranges, too. Foundational to range-ify more of std.path.

Added unit tests for relaxed types.

### CODE REVIEW

File: std/path.d [hunk 1]
```diff
@@ -179,30 +179,122 @@ version (Windows)
...
+    assert(trimDirSeparators("//abc//").array == "abc");
+    assert(trimDirSeparators("//abc//"d).array == "abc"d);
+
+    /* Define a mock BidirectionalRange to use for unittesting.
+     */
+
+    struct BiRange(C)
+    {
+        this(const(C)[] array) { this.array = array; }
+      const
```
Comment (burner) [inline std/path.d]: spacing is off for the next ten lines

**Comment by burner** [comment]
other than the spacing, LGTM

File: std/path.d (hunk 1 above)
Comment (WalterBright) [inline std/path.d]: ok

**Comment by burner** [comment]
BitRange exists twice. IMO there should only be one. On second thought, isn't DummyRange suitable to replace BitRange

**Comment by andralex** [comment]
@WalterBright please consolidate BiRange or reuse DummyRange. After that lgtm.

**Comment by WalterBright** [comment]
consolidated

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**